### ISSUE TITLE
FR: add architecture documents

### ISSUE THREAD

**Issue by obriensystems** [terraform-google-modules/terraform-example-foundation#1177]
### TL;DR

Expand on the architecture sections in the root and folder readme's
https://github.com/terraform-google-modules/terraform-example-foundation?tab=readme-ov-file#terraform-example-foundation

Reference where possible the security foundations architecture that references this TEF repo
https://cloud.google.com/architecture/security-foundations

Reference applied terraform-google-modules and their architecture subset 
for example the project factory https://github.com/terraform-google-modules/terraform-google-project-factory

Ill do a first pass on the default architecture during a reverse engineering of the code - towards the following purposes
- extensibility
- customer deployment documentation
- architecture discussions/modification
- High Value and protected deployments around NIST-800-53, ITSG-33 and FedRAMP 

## Architectural examples include

- https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit/wiki/Architecture
- https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit/blob/dev/solutions/landing-zone/architecture.md
- https://github.com/ssc-spc-ccoe-cei/gcp-documentation/blob/main/Architecture/Repository%20Structure.md
- https://github.com/GoogleCloudPlatform/pbmm-on-gcp-onboarding/blob/main/docs/architecture.md
- https://github.com/Azure/CanadaPubSecALZ/blob/main/docs/architecture.md
- https://aws-samples.github.io/aws-secure-environment-accelerator/latest/architectures/sensitive/
- https://github.com/aws-samples/landing-zone-accelerator-on-aws-for-cccs-medium/tree/main/architecture-doc
- https://github.com/oracle-quickstart/oci-cis-landingzone-quickstart/blob/main/cis-architecture-mapping.md
- https://cloud.google.com/docs/terraform/iac-overview

## Include bidirectional links between this repo and TEF references in
- https://cloud.google.com/docs/terraform/blueprints/terraform-blueprints
- https://cloud.google.com/architecture/security-foundations

### Additional information
- see #1178 

_No response_

**Comment by fmichaelobrien** [comment]
stale bot timer restart - https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/.github/workflows/stale.yml#L21
